source/text/implementation/vss-implementation-utf8_string_handlers.adb

```diff
@@ -352,22 +424,13 @@ package body VSS.Implementation.UTF8_String_Handlers is
             VSS.Implementation.UTF8_Encoding.Encode (Code, L, U1, U2, U3, U4);
 
             if Destination.Capacity < Destination.Size + L then
-               --  Reallocate memory.
-
-               declare
-                  Aux : constant UTF8_String_Data_Access :=
-                    new UTF8_String_Data
-                      (Destination.Capacity
-                       + VSS.Unicode.UTF8_Code_Unit_Count (Item'Length));
-
-               begin
-                  Aux.Storage (Destination.Storage'Range) :=
-                    Destination.Storage;
-                  Aux.Size := Destination.Size;
-                  Aux.Length := Destination.Length;
-                  Self.Unreference (Data);
-                  Destination := Aux;
-               end;
+               --  There is no enough storage to store character, reallocate
+               --  memory.
+
+               Reallocate
+                 (Destination,
+                  VSS.Unicode.UTF8_Code_Unit_Count (Data.Capacity) * 4,
```

Comment:
what is `4` here? Do we grow string size x4 on each Reallocate?

Reply:
It is maximal length of one character in UTF-8 encoding.

Virtual_String.Capacity is minimal number of character that should be able to be stored in the string without memory reallocation (one memory reallocation when capacity is set on non-empty string is allowed when it will be necessary).